sstables/writer.hh

```diff
@@ -52,8 +52,20 @@ public:
     file_writer(file_writer&&) = default;
     // Must be called in a seastar thread.
     void write(const char* buf, size_t n) {
-        _offset.offset += n;
-        _out.write(buf, n).get();
+        try {
+            _out.write(buf, n).get();
+            _offset.offset += n;
+        } catch (...) {
+            auto ep = std::current_exception();
+            try {
+                // close the output_stream so it won't
+                // be destructed while open.
+                _out.close().get();
+            } catch (...) {
+                // ignore nested exception
+            }
+            std::rethrow_exception(ep);
+        }
     }
```

Comment:
I don't think this is the right place to close (after every write). We should prepare a closer after the open (with defer()), since we write in a thread we can use RAII.

Reply:
make sense.  done in v2.